UOP1_Project/Assets/Scripts/SceneManagement/LocationLoader.cs

```diff
@@ -146,24 +146,30 @@ private IEnumerator TrackLoadingProgress()
 			//Iterate through all the scenes to load
 			for (int i = 0; i < _scenesToLoadAsyncOperations.Count; ++i)
 			{
-				Debug.Log("Scene" + i + " :" + _scenesToLoadAsyncOperations[i].isDone + "progress = " + _scenesToLoadAsyncOperations[i].progress);
+				Debug.Log("Scene" + i + ": " + _scenesToLoadAsyncOperations[i].isDone + "\nprogress = " + _scenesToLoadAsyncOperations[i].progress);
 				//Adding the scene progress to the total progress
 				totalProgress += _scenesToLoadAsyncOperations[i].progress;
 			}
 			//The fillAmount for all scenes, so we devide the progress by the number of scenes to load
 			_loadingProgressBar.fillAmount = totalProgress / _scenesToLoadAsyncOperations.Count;
-			Debug.Log("progress bar" + _loadingProgressBar.fillAmount + "and value =" + totalProgress / _scenesToLoadAsyncOperations.Count);
+			Debug.Log("progress bar " + _loadingProgressBar.fillAmount + " and value = " + totalProgress / _scenesToLoadAsyncOperations.Count);
 
 			yield return null;
 		}
 
 		//Clear the scenes to load
-		_scenesToLoadAsyncOperations.Clear();
+		OnLoadingFinished();
 
 		//Hide progress bar when loading is done
 		_loadingInterface.SetActive(false);
 	}
 
+	private void OnLoadingFinished()
```

Comment:
The name of this function is really misleading, because it happens when the total loading is done in case it needs to show a progress bar (which would make sense), but it's also executed as soon as the loading is starting if the loading screen is not needed.